### PR TITLE
Updated AWS SDK dependency from 1.x to SDK 2.15

### PR DESCRIPTION
**Pull request by bhouse99**
Updated to support AWS SDK 2 to better support later systems.

Created tag marking AWS SDK 1 dependency in case this causes
a down stream issue.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # IMPORTANT: Latest Version
 
-The current version is 4.0.3. Please see the [changelog](./CHANGELOG.md) for details on version history.
+The current version is 4.0.4. Please see the [changelog](./CHANGELOG.md) for details on version history.
```
Comment (dchenbec) [inline README.md:3]: Can we also update the section titled "Add the Authentication Plugin to the Application" below?

**Review by dchenbec — CHANGES_REQUESTED** [review]
Looks good with a minor nit

File: README.md (hunk 1 above)
Comment (bhouse99) [reply README.md:3]: Ok will do

**Review by dchenbec — APPROVED** [review]
Looks good to me

**Pull request merged**